src/CodeGenerator/AutoloadGenerator.php

```diff
@@ -7,87 +7,94 @@
 
 namespace Zend\Di\CodeGenerator;
 
-use Zend\Code\Generator\DocBlockGenerator;
-use Zend\Code\Generator\FileGenerator;
-use Zend\Code\Generator\ClassGenerator;
-use Zend\Code\Generator\MethodGenerator;
-use Zend\Code\Generator\PropertyGenerator;
-use Zend\Code\Generator\PropertyValueGenerator;
+use SplFileObject;
+use Throwable;
+use Zend\Di\Exception\GenerateCodeException;
+
+use function array_keys;
+use function array_map;
+use function file_get_contents;
+use function implode;
+use function str_repeat;
+use function strtr;
+use function var_export;
 
 class AutoloadGenerator
 {
     use GeneratorTrait;
 
-    private $namespace;
+    private const CLASS_TEMPLATE = __DIR__ . '/../../templates/autoloader-class.template';
+    private const FILE_TEMPLATE = __DIR__ . '/../../templates/autoloader-file.template';
 
     /**
-     * @param string $namespace
+     * @var string
      */
+    private $namespace;
+
     public function __construct(string $namespace)
     {
         $this->namespace = $namespace;
     }
 
-    private function generateAutoloaderClass(array &$classmap)
+    private function writeFile(string $filename, string $code) : void
+    {
+        try {
+            $file = new SplFileObject($filename, 'w');
+            $file->fwrite($code);
+        } catch (Throwable $e) {
+            throw new GenerateCodeException(sprintf('Failed to write output file "%s"', $filename), 0, $e);
+        }
+    }
+
+    private function buildFromTemplate(string $templateFile, string $outputFile, array $replacements) : void
+    {
+        $template = file_get_contents($templateFile);
+        $code = strtr($template, $replacements);
+        $outputFile = $this->outputDirectory . '/' . $outputFile;
+
+        $this->writeFile($outputFile, $code);
+    }
+
+    private function generateClassmapCode(array &$classmap) : string
     {
-        $class = new ClassGenerator('Autoloader');
-        $classmapValue = new PropertyValueGenerator(
-            $classmap,
-            PropertyValueGenerator::TYPE_ARRAY_SHORT,
-            PropertyValueGenerator::OUTPUT_MULTIPLE_LINE
+        $lines = array_map(
+            function (string $class, string $file): string {
+                return var_export($class, true) . ' => ' . var_export($file, true) . ',';
```

Comment:
This might benefit from `sprintf` as well:

```php
return sprintf(
    '%s => %s,',
    var_export($class, true),
    var_export($file, true)
);
```

When I first read it, it read like a `yield` statement, and I missed the concatenation.